src/QsCompiler/Transformations/TypeResolutionCombination.cs

```diff
@@ -282,33 +282,46 @@ public override ResolvedTypeKind OnTypeParameter(QsTypeParameter tp)
 
         /// <summary>
         /// Walker that checks a given type parameter resolution to see if it constricts
-        /// the type parameter to another type parameter of the same callable.
+        /// the type parameter to another type parameter of the same callable or to a
+        /// nested reference of the same type parameter.
         /// </summary>
         private class CheckForConstriction : TypeTransformation<CheckForConstriction.TransformationState>
         {
-            private readonly TypeParameterName typeParamName;
-
             /// <summary>
             /// Walks the given ResolvedType, typeParamRes, and returns true if there is a reference
-            /// to a different type parameter of the same callable as the given type parameter, typeParam.
+            /// to a different type parameter of the same callable as the given type parameter, typeParam,
+            /// or the same type parameter but in a nested type.
             /// Otherwise returns false.
             /// </summary>
-            public static bool Apply(TypeParameterName typeParam, ResolvedType typeParamRes)
+            public static bool IsConstrictiveResolution(TypeParameterName typeParam, ResolvedType typeParamRes)
```

Comment:
This is a part of the public API and it hence cannot simply be renamed. Since this is a bug fix, it also does not change the behavior in a way that would warrant deprecating it in favor of a new name.

Reply:
This is part of a private class, so it isn't part of the public API.

Reply:
Sorry, I only looked at the diff.